sda-commons-starter/src/main/java/org/sdase/commons/starter/SdaPlatformBundle.java

```diff
@@ -0,0 +1,352 @@
+    bootstrap.addBundle(ConfigurationSubstitutionBundle.builder().build());
+    bootstrap.addBundle(DefaultLoggingConfigurationBundle.builder().build());
+    bootstrap.addBundle(InternalHealthCheckEndpointBundle.builder().build());
+    bootstrap.addBundle(JaegerBundle.builder().build());
+    bootstrap.addBundle(OpenTracingBundle.builder().build());
+    bootstrap.addBundle(PrometheusBundle.builder().build());
+    bootstrap.addBundle(TraceTokenBundle.builder().build());
+
+    // add configured bundles
+    List<ConfiguredBundle<? super C>> configuredBundles = new ArrayList<>();
+    configuredBundles.add(jacksonConfigurationBundleBuilder.build());
+    configuredBundles.add(securityBundleBuilder.build());
+    configuredBundles.add(openApiBundleBuilder.build());
+    if (authBundleBuilder != null) {
+      configuredBundles.add(authBundleBuilder.build());
+    }
+    if (opaBundleBuilder != null) {
+      configuredBundles.add(opaBundleBuilder.build());
+    }
+    if (corsBundleBuilder != null) {
+      configuredBundles.add(corsBundleBuilder.build());
+    }
+    if (consumerTokenBundleBuilder != null) {
+      configuredBundles.add(consumerTokenBundleBuilder.build());
+    }
+    configuredBundles.stream().map(b -> (ConfiguredBundle) b).forEach(bootstrap::addBundle);
+  }
+
+  @Override
+  public void run(C configuration, Environment environment) {
+    // not needed for the platform bundle, created bundles are added in initialize
+  }
+
+  public static class InitialBuilder<C extends Configuration>
+      implements InitialPlatformBundleBuilder,
+          AuthConfigProviderBuilder<C>,
+          CorsConfigProviderBuilder<C>,
+          ConsumerTokenConfigBuilder<C>,
+          ConsumerTokenRequiredConfigInitialBuilder<C>,
+          OpenApiInitialBuilder<C>,
+          OpenApiFinalBuilder<C>,
+          PlatformBundleBuilder<C> {
+
+    private AuthBundle.AuthBuilder<C> authBundleBuilder;
+    private OpaBundle.OpaBuilder<C> opaBundleBuilder;
+    private ConsumerTokenConfig consumerTokenConfig;
+    private ConsumerTokenBundle.FinalBuilder<C> consumerTokenBundleBuilder;
+    private SecurityBundle.Builder securityBundleBuilder = SecurityBundle.builder();
+    private JacksonConfigurationBundle.Builder jacksonBundleBuilder =
+        JacksonConfigurationBundle.builder();
+    private CorsBundle.FinalBuilder<C> corsBundleBuilder;
+    private OpenApiBundle.FinalBuilder openApiBundleBuilder;
+
+    private InitialBuilder() {}
+
+    // Final step
+
+    @Override
+    public SdaPlatformBundle<C> build() {
+      return new SdaPlatformBundle<>(
+          securityBundleBuilder,
+          jacksonBundleBuilder,
+          authBundleBuilder,
+          opaBundleBuilder,
+          corsBundleBuilder,
+          consumerTokenBundleBuilder,
+          openApiBundleBuilder);
+    }
+
+    // InitialBuilder
+
+    @Override
+    public ConsumerTokenConfigBuilder<SdaPlatformConfiguration> usingSdaPlatformConfiguration() {
```

Comment:
Open for discussion: Since most applications derive their configuration from `SdaPlatformConfiguration` it would be nice to give them a shortcut like `public <T extends SdaPlatformConfiguration> AuthConfigProviderBuilder<T> usingSdaPlatformConfiguration(Class<T> configurationClass)`. I would also like to enable OPA by default in this builder. No OPA would require to use the `usingCustomConfig()` branch.

Reply:
I've created a PR for this change. Let's discuss the suggestion there. 
[https://github.com/SDA-SE/sda-dropwizard-commons/pull/654](https://github.com/SDA-SE/sda-dropwizard-commons/pull/654)